### PR TITLE
[16.04] Fix dbkey problem discovered by @hrhotz

### PR DESCRIPTION
**Pull request by jmchilton**
More info here #2795 and http://dev.list.galaxyproject.org/problems-with-action-type-quot-metadata-quot-name-quot-dbkey-quot-in-tool-xml-td4669886.html.

### CODE REVIEW

File: test/functional/tools/dbkey_output_action.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            <actions>
+                <action type="metadata" name="dbkey">
+                    <option type="from_data_table" name="test_fasta_indexes" column="1" offset="0">
+                        <filter type="param_value" column="0" value="#" compare="startswith" keep="False"/>
+                        <filter type="param_value" ref="index" column="0"/>
+                    </option>
+                </action>
+            </actions>
+        </data>
+    </outputs>
```
Comment (nsoranzo) [inline test/functional/tools/dbkey_output_action.xml:23]: Without a `<tests>` section this is not really testing that metadata assignment works, is it?

See https://github.com/galaxyproject/galaxy/pull/2795/files#r74952233 for a test template.

Comment (jmchilton) [inline test/functional/tools/dbkey_output_action.xml:23]: No - it is included for manual testing purposes and as demonstration. If I add that template will it work?

File: lib/galaxy/tools/actions/__init__.py [hunk 2]
```diff
@@ -628,7 +628,7 @@ def set_metadata_defaults( self, output, dataset, tool, on_text, trans, incoming
         """
         if output.actions:
             for action in output.actions.actions:
-                if action.tag == "metadata":
+                if action.tag == "metadata" and action.default:
                     metadata_new_value = fill_template( action.default, context=params ).split(",")
```
Comment (nsoranzo) [inline lib/galaxy/tools/actions/__init__.py:632]: Should we also check inside `galaxy.util.template.fill_template()` that `template_text` is not None, or is there a legitimate use case?

Comment (jmchilton) [inline lib/galaxy/tools/actions/__init__.py:632]: ... I don't know?

Comment (nsoranzo) [inline lib/galaxy/tools/actions/__init__.py:632]:
```
from galaxy.util.template import fill_template

fill_template(None, dict())
```

return the string `'<Cheetah.Template.Template object at 0x7f3bec07d890>'`, so I don't think there could be any use case.

Comment (jmchilton) [inline lib/galaxy/tools/actions/__init__.py:632]: Cool - I'll add an exception

File: test/functional/tools/dbkey_output_action.xml (hunk 1 above)
Comment (nsoranzo) [inline test/functional/tools/dbkey_output_action.xml:23]: Not without input and output files, test data table and loc file.

Comment (jmchilton) [inline test/functional/tools/dbkey_output_action.xml:23]: Updated with valid values - it seems to work. It should show up as a new test under Jenkin's framework tests now. Thanks for pushing me on this.

File: lib/galaxy/tools/actions/__init__.py (hunk 2 above)
Comment (jmchilton) [inline lib/galaxy/tools/actions/__init__.py:632]: Done. I'd probably do this in dev instead but we will see if anyone complains.

Comment (nsoranzo) [inline lib/galaxy/tools/actions/__init__.py:632]: Perfect, thanks!

**Comment by nsoranzo** [comment]
:+1:

**Comment by hrhotz** [comment]
brilliant! Thank you vcery much for the fix!

**Comment by mvdbeek** [comment]
👍

**Comment by mvdbeek** [comment]
Thanks @jmchilton @nsoranzo @hrhotz!

**Pull request merged**